### PR TITLE
some cleanup and fix #14

### LINKED ISSUE #14: ocaml.type support for compound types
```
$ cat q.proto 
type string_set = [ string ] options "ocaml.type" = "SS.t, SS.of_list, SS.elements"
message x = { set : string_set; }
```

```
$ extprotc q.proto -o /dev/stdout | grep SS
    type string_set = SS.t;;
      Extprot.Pretty_print.pp_list Extprot.Pretty_print.pp_string ppf (SS.elements x);;
    type _x = { set : SS.t };;
    type x = { set : SS.t };;
```
So it doesn't use conversion functions when reading/writing value..

### PR DESCRIPTION
**Pull request by ygrek**
- fix omake warning
- get rid of some deprecated `String` functions and add dependency on `bytes` (code already needed this before changes)
- fix "unused value" warnings by prefixing names with underscore, please review if any of those names _should_ have been used
- notably the above fixes potential bug of merging type options for `Record` in `gencode.ml`
- amazingly that same function also had a cause of bug #14 (unfortunately compiler could not help here)

### CODE REVIEW

**Comment by mfp** [comment]
Thank you for the bugfix + overall cleanup.

I've reviewed the patch and there's only 1 place where the value could possibly be used (record type expansion in gencode.ml, allowing to attach options to a "record type application" in a message definition if the parser were updated to accept them), and another where I'm not sure why there's a different behavior based on the existence of a default value (field reading in gen_ocaml.ml).

At any rate AFAICS the patch doesn't introduce any new bugs, so I'm merging and flagging the suspicious locations so that they're not forgotten :)

**Comment by ygrek** [comment]
just pushed fixes for `omake test` also, looks like they slipped past the merge :)

File: compiler/gen_OCaml.ml [hunk 1]
```diff
@@ -888,7 +888,7 @@ struct
                   ~field:$str:name$
                   e loc
             >>
-        | Some expr ->
```
Comment (mfp) [inline compiler/gen_OCaml.ml]: Not sure why there's a separate branch for the case where a default value exists, where deliberately only `Bad_wire_type` errors are caught.

**Comment by mfp** [comment]
On Mon, Feb 01, 2016 at 11:37:23AM -0800, ygrek wrote:
> just pushed fixes for `omake test` also, looks like they slipped past the merge :)

Hmm that's embarrassing... Turns out I had dummy_type.ml in my working dir
since Nov 2014 and forgot to add it in 5c820d3 :-/

-- 
Mauricio Fernández

**Pull request merged**